sympy/matrices/expressions/tests/test_matexpr.py

```diff
@@ -405,38 +405,15 @@ def test_MatMul_postprocessor():
     assert Mul(A, x, M, M, x) == MatMul(A, Mx**2)
 
 
-@XFAIL
-def test_MatAdd_postprocessor_xfail():
-    # This is difficult to get working because of the way that Add processes
-    # its args.
-    z = zeros(2)
-    assert Add(z, S.NaN) == Add(S.NaN, z)
-
-
 def test_MatAdd_postprocessor():
-    # Some of these are nonsensical, but we do not raise errors for Add
-    # because that breaks algorithms that want to replace matrices with dummy
-    # symbols.
```

Comment:
I'm not sure how many of these are covered by the tests. I know it came up in situations of trying to call various simplification functions on matrix expressions. It might be worth checking a few to make sure there aren't any regressions here. See https://github.com/sympy/sympy/pull/15948 for the nitty-gritty details.